src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs

```diff
@@ -124,14 +124,25 @@ public class SampleTemperatureData
             public float Temperature { get; set; }
         }
 
-        public static IEnumerable<SampleTemperatureData> GetSampleTemperatureData()
+        /// <summary>
+        /// Get a fake temperature dataset.
+        /// </summary>
+        /// <param name="exampleCount">The number of examples to return.</param>
+        /// <returns>An enumerable of <see cref="SampleTemperatureData"/>.</returns>
+        public static IEnumerable<SampleTemperatureData> GetSampleTemperatureData(int exampleCount)
```

Comment:
>int exampleCount [](start = 82, length = 16)

I suggest using 'yield return' for creating the lazy IEnumerable and removing the sampleCount argument. We're reinventing Take() in Linq, which does exactly this.

Reply:
This is a style suggestion though. I'll leave it up to you.

---
In reply to: [253624396](https://github.com/dotnet/machinelearning/pull/2405#discussion_r253624396) [](ancestors = 253624396)

Reply:
Good call on the `yield return`.